src/inference/hmc.jl

```diff
@@ -420,9 +417,7 @@ function AbstractMCMC.step!(
     # Get position and log density before transition
     θ_old, log_density_old = spl.state.vi[spl], getlogp(spl.state.vi)
     if spl.selector.tag != :default
-        update_hamiltonian!(spl, model, length(θ_old))
-        resize!(spl.state.z.θ, length(θ_old))
-        spl.state.z.θ .= θ_old
+        update_hamiltonian!(rng, spl, model, θ_old)
```

Comment:
I think the old code was just fine, no?

Reply:
For Gibbs I believe it's needed.

Reply:
Yes I think you are right because we need the logp at the current point to compute the correct acceptance ratio of the proposal I suppose.

Reply:
But we can just input that from `vi` so maybe we don't need to re-run the logp function and its gradient.

Reply:
But this is actually done inside `step` where `refresh` is called. So it's definitely not needed :)

Reply:
Cool. I'm unware of if this is perfomed already somewhere else. But I will keep it in mind in the future. I will also try to think about how to prevent this from AHMC side if possible as it causes us a few times of issues now.